### PR TITLE
Upgrade Python to 3.9 with pyupgrade.

### PR DESCRIPTION
**Pull request by bbannier**

### CODE REVIEW

File: tests/Scripts/license-header.py [hunk 1]
```diff
@@ -6,10 +6,8 @@
 for f in sys.argv[1:]:
     with open(f) as input:
         if not any(
```
Comment (timwoj) [inline tests/Scripts/license-header.py:8]: Huh, I'd never seen that before.

File: tests/Scripts/stray_baselines.py [hunk 2]
```diff
@@ -10,14 +10,12 @@
...
-        available_tests = set(
-            map(
-                lambda x: x.decode("utf-8"),
-                subprocess.check_output(
-                    ["btest", "-l", "-c", TEST_DIR + "/btest.cfg"]
-                ).splitlines(),
-            )
-        )
+        available_tests = {
+            x.decode("utf-8")
```
Comment (timwoj) [inline tests/Scripts/stray_baselines.py:14]: What exactly is happening with this `x.decode` here? `x` isn't set to anything, so it's not just decoding a string.

**Review by timwoj — COMMENTED** [review]
Huh, I'd never seen that `if not any` construct. Looks fine to me.

File: tests/Scripts/stray_baselines.py (hunk 2 above)
Comment (bbannier) [reply tests/Scripts/stray_baselines.py:14]: The part in `{..}` is a set comprehension, `x` is bound via `for x in ...` on the line below.

File: tests/Scripts/license-header.py (hunk 1 above)
Comment (bbannier) [reply tests/Scripts/license-header.py:8]: Already before this was more functional style; the arg to `any` is new a generator expression here. I personally find this still pretty readable, but I could rewrite it e.g., using a temporary list if you could that would help.

**Review by timwoj — APPROVED** [review]

**Pull request merged**